### PR TITLE
tracker domain map

### PR DESCRIPTION
**Pull request by russellholt**
Add a tracker domain map to the companyList importer as a convenient reverse mapping from domain to company name and tracker type, for the top tracker networks. It's called `TopTrackerDomains` in data/tracker_lists/trackersWithParentCompany, sibling to the type categories. Also fixes Amazon.com naming to 'Amazon'.

It looks like:

```
    "TopTrackerDomains": {
...
        "estara.com": {
            "c": "Oracle",
            "t": "Advertising"
        },
        "alexa.com": {
            "c": "Amazon",
            "t": "Analytics"
        },
        "alexametrics.com": {
            "c": "Amazon",
            "t": "Analytics"
        },
        "facebook.com": {
            "c": "Facebook",
            "t": "Social"
        },
...
```

For the Grade Detail subview, this lets you map domains like `alexametrics.com` to 'Amazon' (major tracker network it is part of) and 'Analytics' (tracker name).

This is not very big, ~ 60 domains for now and not likely to get much bigger for this purpose, ie domains owned by top tracker networks and their categories won't every be very large.

We could:

- use this as is and do reverse mapping with domains on the Grade Details subview.
- take this ~ 60 entries and for now simply hard code them into the view that needs it, and work out the proper importer later
- save it as a separate file which could be require()d wherever needed.

## Steps to test this PR:
1. Run grunt build or dev,
2. Examine data/tracker_lists/trackersWithParentCompany 
3. Decide what to do with it

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: scripts/importers/companyList.js [hunk 1]
```diff
@@ -5,11 +5,13 @@ global.companyList = function(listData){
      */
     var companyListLoc = 'https://raw.githubusercontent.com/mozilla-services/shavar-prod-lists/master/disconnect-blacklist.json';
     var remapDataLoc = 'https://raw.githubusercontent.com/mozilla-services/shavar-prod-lists/master/google_mapping.json'
-    var trackerList = {};
+    var trackerList = { TopTrackerDomains:{}};
     var trackerTypes = ['Advertising', 'Analytics', 'Disconnect', 'Social'];
     var request = require('request');
     var remapData, companyList;
     
+    let majorNetworks = {"Google":true, "Facebook":true, "Twitter":true, "Amazon":true, "AppNexus":true, "Oracle":true};
```
Comment (jdorweiler) [inline scripts/importers/companyList.js]: you can get this from defaultSettings. https://github.com/duckduckgo/chrome-zeroclickinfo/blob/beta/data/defaultSettings.js

Comment (russellholt) [reply scripts/importers/companyList.js]: is settings really the right place for that?

Comment (russellholt) [reply scripts/importers/companyList.js]: either way, @jdorweiler  and I agreed on a separate discussion to ignore this for now as the settings file is js and declares a const. Could be changed but needs to be fixed in another PR.

Comment (russellholt) [reply scripts/importers/companyList.js]: added a note in the source

Comment (jdorweiler) [reply scripts/importers/companyList.js]: 👍

**Comment by laurengarcia** [comment]
Picking this branch up...

**Comment by laurengarcia** [comment]
@russellholt @jdorweiler So if you pull this branch down and run `grunt build` you'll see the `majorTrackingNetworks` are now defined in defaultSettings and `data/tracker_lists/trackersWithParentCompany.json` now has the additional field TopTrackerDomains, however in site.js we also have the Princeton data hard-coded and referred to as 'majorTrackingNetwork', which seems to conflict here. 

Am unsure of what the intention is supposed to be here; seems like we should clean up hard-coded data? or come up with another definition vs. 'majorTrackingNetworks' - ? I'm also not entirely sure what the intention is here. Needs clarity.

**Comment by laurengarcia** [comment]
Okay, phew. This came out pretty nice in the end. Ended up using the reverse mapping after all, am requiring it as a dependency directly from /data dir into relevant ui model. Just wish we had more of this data now!

tracker-categories1

tracker-categories2

**Comment by laurengarcia** [comment]
Looks like Jason approved changes/updates here, going to merge.

**Pull request merged**